lib/reservations.rb

```diff
@@ -0,0 +1,40 @@
+require 'date'
+
+module Hotel
+  class Reservation
+    ROOMCOST = 200
+    attr_accessor :check_in, :check_out, :room_num
+    #things that only have to deal with SINGLE ROOM, one reservation at a time
+    #monitors overlapping
+    def initialize(check_in, check_out, room_num)
+      @check_in = check_in
+      @check_in = check_out
+      @room_num = room_num
+        if @room_num > 20 || @room_number < 1
+          raise ArgumentError.new("Number must be between 1 - 20")
+        end
+    end
+
+    def reserve_room(check_in, check_out, room_num)
+      #reserves a room for a given date range
+      in = Date.parse(check_in)
+      out = Date.parse(check_out)
+      @range = (out - in)
+
+      if @reservations[@rooms.last] == room_num
```

Comment:
I'm not sure what this `if` statement is trying to accomplish.  I think what you want to do is loop through all your reservations and check to see if:

1.  The reservation has the same room number **and**
1.  The dates conflict with the dates given as arguments.

If none do, then you can create a **new** reservation instance and add that to the list of reservations.  Otherwise indicate an error.  

However this really isn't the job of an individual `Reservation` instance (the method here is in the wrong class).  Instead making reservations is a job of some kind of booking manager.  A `Reservation` class should describe all the things you could ask an individual reservation (does it conflict with a given date range, what's the room #, how much does it cost etc).